tests/specs/lemmas.k

```diff
@@ -415,4 +415,12 @@ module LEMMAS-HASKELL [symbolic, kore]
 
     rule B ==Bool false => notBool B [simplification]
 
+    rule #range(A ++ _B, START, LENGTH) => #range(A, START, LENGTH)
+      requires START +Int LENGTH <=Int #sizeByteArray(A) [simplification]
+
+    rule #range(A ++ B, START, LENGTH) => #range(B, START -Int #sizeByteArray(A), LENGTH)
+      requires #sizeByteArray(A) <=Int START [simplification]
+
+    rule #range ( #buf ( LEN , VALUE ) , 0 , LEN ) => #buf (LEN, VALUE) [simplification]
```

Comment:
```suggestion
    rule #range ( BA:ByteArray , 0 , LEN ) => BA requires #sizeByteArray(BA) ==Int LEN [simplification]
```

Does this work? It seems more general.

Reply:
Checking it locally now, will add your suggestion if it works.